src/openklant/management/commands/migrate_to_v2.py

```diff
@@ -0,0 +1,176 @@
+# SPDX-License-Identifier: EUPL-1.2
+# Copyright (C) 2024 Dimpact
+
+
+import logging
+import os
+from dataclasses import asdict, fields as dataclass_fields
+from typing import Any, Tuple
+from urllib.parse import parse_qs, urlparse
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured, ValidationError
+from django.core.management import CommandError
+from django.core.management.base import BaseCommand, CommandParser
+
+from rest_framework.fields import URLValidator
+from rest_framework.reverse import reverse_lazy
+
+from openklant.components.token.models import TokenAuth
+from openklant.migration.client import Client
+from openklant.migration.v1.client import LegacyOpenKlantClient
+from openklant.migration.v1.data import Klant
+from openklant.migration.v2.client import OpenKlantClient
+
+logger = logging.getLogger(__name__)
+
+
+KLANT_FIELDS = {field.name: field for field in dataclass_fields(Klant)}
+
+DIGITALE_ADDRESSEN_PATH = reverse_lazy(
+    "klantinteracties:digitaaladres-list",
+    kwargs=dict(version=settings.REST_FRAMEWORK["DEFAULT_VERSION"]),
+)
+
+PARTIJEN_PATH = reverse_lazy(
+    "klantinteracties:partij-list",
+    kwargs=dict(version=settings.REST_FRAMEWORK["DEFAULT_VERSION"]),
+)
+
+
+def _retrieve_klanten(url: str, access_token: str) -> Tuple[list[Klant], str | None]:
+    klanten_path = "/klanten/api/v1/klanten"
+
+    _url = urlparse(url)
+    _params = parse_qs(_url.query)
+
+    client = LegacyOpenKlantClient(f"{_url.scheme}://{_url.netloc}", access_token)
+    response_data = client.retrieve(_url.path or klanten_path, params=_params)
+
+    if not isinstance(response_data, dict):
+        logger.error(f"Unexpected response data returned: {response_data}")
+        return [], None
+
+    items = response_data.get("results", [])
+    klanten = []
+
+    generic_client = Client()
+
+    for data in items:
+        klant = Klant(
+            **{field: value for field, value in data.items() if field in KLANT_FIELDS}
+        )
+
+        if klant.subject and not klant.subject_type:
+            subject_data = generic_client.retrieve(klant.subject)
+
+            if not isinstance(subject_data, dict):
+                logger.error(
+                    "Unexpected response data returned during retrieval of "
+                    f"subject: {subject_data}"
+                )
+
+                continue
+
+            klant.set_from_external_subject(subject_data)
+
+        klanten.append(klant)
+
+    next_url = response_data.get("next")
+    return klanten, next_url
+
+
+def _save_klanten(url: str, klanten: list[Klant]) -> list[str]:
+    if not klanten:
+        logger.info("No klanten to save, moving on...")
+        return []
+
+    logger.info(f"Trying to create {len(klanten)} klanten through the V2 API")
+
+    dummy_token = _generate_dummy_token()
```

Comment:
If I understand this correctly, it if the list is paginated and `_save_klanten` is run multiple times, you'll have multiple dummy tokens. It might be better to call this outside the loop in `.handle` to avoid this.

Reply:
While it used `get_or_create` to retrieve the token before changes were made to this, I agree that querying the database multiple times is unnecessary, see https://github.com/maykinmedia/open-klant/pull/263/commits/293156ee77fa8ee876d8d0d92053eaf9e6cd1c7f.